x-pack/plugins/ml/server/models/job_validation/validate_time_range.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ES_FIELD_TYPES } from '../../../../../../src/plugins/data/server';
+import { parseInterval } from '../../../../../legacy/plugins/ml/common/util/parse_interval';
+import { CombinedJob } from '../../../../../legacy/plugins/ml/public/application/jobs/new_job/common/job_creator/configs';
+import { callWithRequestType } from '../../../../../legacy/plugins/ml/common/types/kibana';
+// @ts-ignore
+import { validateJobObject } from './validate_job_object';
+
+interface ValidateTimeRangeMessage {
+  id: string;
+  timeField?: string;
+  minTimeSpanReadable?: string;
+  bucketSpanCompareFactor?: number;
+}
+
+interface TimeRange {
+  start: number;
+  end: number;
+}
+
+const BUCKET_SPAN_COMPARE_FACTOR = 25;
+const MIN_TIME_SPAN_MS = 7200000;
+const MIN_TIME_SPAN_READABLE = '2 hours';
+
+export async function isValidTimeField(callWithRequest: callWithRequestType, job: CombinedJob) {
```

Comment:
this should really be changed to `callAsCurrentUser: APICaller` for when the files that call these functions get typescriptified.

Reply:
Done in https://github.com/elastic/kibana/pull/59017/commits/b92491bcdc167e2a2b546286e0774e713e60abf3